WooCommerce/src/main/kotlin/com/woocommerce/android/analytics/AnalyticsTracker.kt

```diff
@@ -22,11 +26,24 @@ class AnalyticsTracker private constructor(
     private val selectedSite: SelectedSite,
     private val appPrefs: AppPrefs,
     private val getWooVersion: GetWooCorePluginCachedVersion,
+    appCoroutineScope: CoroutineScope,
 ) {
     private var tracksClient: TracksClient? = TracksClient.getClient(context)
     private var username: String? = null
     private var anonymousID: String? = null
 
+    private val analyticsEventsToTrack = Channel<Pair<AnalyticsEvent, Map<String, *>>>(capacity = BUFFERED)
+
+    init {
+        appCoroutineScope.launch {
+            for (event in analyticsEventsToTrack) {
+                doTrack(event.first, event.second)
+            }
```

Comment:
Just a minor np in case you want to add a more producer-consumer style: 
```suggestion
            analyticsEventsToTrack.consumeEach { (stat, properties) ->
                doTrack(stat, properties)
            }
```

Reply:
Looks better indeed. Thanks!